### ISSUE TITLE
`DTZ007` with F-string format specifier

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ruff#10601]
Don't know if this is the same issue, but there is also a false-positive if the format string is an f-string.

For example:
```python
from datetime import datetime
def parse_iso(iso_str,millis=True):
    return datetime.strptime(iso_str, f"%Y-%m-%dT%H:%M:%S{('.%f' if millis else '')}%z")
```

Even without a condition inside the f-string, it immediately triggers the lint error.

For example:
```python
from datetime import datetime
dt = datetime.strptime("","%Y-%m-%d %H:%M:%S%z") # everything okay

dt = datetime.strptime("", f"%Y-%m-%d %H:%M:%S%z") # throws DTZ007
```

_Originally posted by @jonas-w in https://github.com/astral-sh/ruff/issues/1306#issuecomment-2018928777_
